### PR TITLE
Mainnet Support

### LINKED ISSUE #124: Support mainnet testing where possible

### PR DESCRIPTION
**Pull request by JakeUrban**
resolves stellar/transfer-server-validator#124

Adds the ability for tests to run on mainnet. 

The only section that required any adjustments for mainnet testing was the SEP-10 tests, as they were the only ones that required accounts to be created on the network.

- Adds 'Run on mainnet' checkbox to UI
- Added check for correct network before running tests
- Added temp account creation from master account specified by env var
- Added temp account merging to master account once tests finish

### CODE REVIEW

File: client/src/App.tsx [hunk 1]
```diff
@@ -14,23 +15,33 @@ function App() {
...
+  const [wrongNetworkError, setWrongNetworkError] = useState<String | null>(
+    null,
+  );
   const [domain, setDomain] = useState("testanchor.stellar.org");
   const [currency, setCurrency] = useState<string>("");
   const [runOptionalTests, setRunOptionalTests] = useState<boolean>(
     Boolean(parseInt(process.env.RUN_OPTIONAL_TESTS || "0")) || false,
   );
+  const [runOnMainnet, setRunOnMainnet] = useState<boolean>(
+    Boolean(parseInt(process.env.MAINNET || "0")) || false,
```
Comment (msfeldstein) [inline client/src/App.tsx]: Does the app frontend have access to process.env?

File: client/src/App.tsx [hunk 2]
```diff
@@ -44,6 +55,22 @@ function App() {
   }, [availableTests]);
   useEffect(resetTests, [availableTests]);
 
+  const setDomainArgs = useCallback(() => {
+    if (currency) {
+      window.history.replaceState(
+        null,
+        "",
+        `?home_domain=${domain}&currency=${currency}&mainnet=${runOnMainnet}`,
```
Comment (msfeldstein) [inline client/src/App.tsx]: Need to add `project` param in here now that the project switcher landed

File: client/src/App.tsx [hunk 3]
```diff
@@ -61,6 +88,60 @@ function App() {
     changeOptionalTestStatuses();
   }, [runOptionalTests]);
 
+  /*
+   * Checks the anchor's TOML for NETWORK_PASSPHRASE and validates that
+   * its the correct string depending on runOnMainnet
+   */
+  const checkAnchorNetwork = useCallback(async () => {
+    let url = domain.includes("http") ? domain : "https://" + domain;
```
Comment (msfeldstein) [inline client/src/App.tsx]: Can we reuse the 'getValidDomain' function below so we don't duplicate this logic?

File: client/src/App.tsx (hunk 1 above)
Comment (JakeUrban) [reply client/src/App.tsx]: Maybe its different in production but in development the server and client are launched from the same parent process so I think so.

**Comment by JakeUrban** [comment]
I need to merge this with the latest master since we merged the select-SEP functionality

**Pull request merged**